lib/pure/algorithm.nim

```diff
@@ -563,7 +563,8 @@ proc product*[T](x: openArray[seq[T]]): seq[seq[T]] =
   ## Produces the Cartesian product of the array.
   ## Every element of the result is a combination of one element from each seq in `x`,
   ## with the ith element coming from `x[i]`.
-  ## Warning: complexity may explode.
+  ## 
+  ## .. warning:: complexity may explode.
```

Comment:
```suggestion
  ## .. warning:: Use it with caution. Complexity may explode.
```

What complexity tho?, Cyclomatic complexity ?, O-notation performance?.

Reply:
* we should be precise here:
```nim
## .. warning:: this returns `x[0].len * ... * x[^1].len` elements, which can use a lot of memory
```
* also, I don't like this proc, instead (in future work) we should deprecate it in favor of https://github.com/timotheecour/Nim/issues/633